controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -895,10 +901,6 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 				info.log.Info("failed creating hostfirmwarecomponents")
 				return actionError{errors.Wrap(err, "failed creating hostFirmwareComponents")}
 			}
-			if err = r.hostUpdatePolicySetOwnerReference(info); err != nil {
-				info.log.Info("failed setting owner reference on hostupdatepolicy")
-				return actionError{errors.Wrap(err, "failed setting owner reference on hostUpdatePolicy")}
-			}
```

Comment:
I think we want to continue to do this as early as possible (i.e. during registration).

Reply:
You mean adding a call for `acquireHostUpdatePolicy` here correct? (since we changed the name `hostUpdatePolicySetOwnerReference` and we only call in `managePower`. 
The HostUpdatePolicy is not automatically created, so I'm wondering if we really need during registration 🤔 
@dtantsur do you have thoughts on it?

Reply:
added
```
			if _, err = r.acquireHostUpdatePolicy(info); err != nil {
				info.log.Info("failed setting owner reference on hostupdatepolicy")
				return actionError{errors.Wrap(err, "failed setting owner reference on hostUpdatePolicy")}
			}
```

Reply:
I think it's a fair call: we need to ensure that even user-created objects have the right owner reference, and we should not wait with that until the power action.